src/hooks/events.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { createContext, useContext, useMemo } from "react";
+import { useAsync } from "react-use";
+
+import { fetchAllVenueEvents } from "api/events";
+
+import { ReactHook } from "types/utility";
+import { VenueEvent } from "types/venues";
+
+import { WithId, WithVenueId } from "utils/id";
+import { tracePromise } from "utils/performance";
+import { isTruthy } from "utils/types";
+
+import { useSovereignVenueId } from "./useSovereignVenueId";
+
+const emptyArray: never[] = [];
+
+export interface VenueEventsProps {
+  venueIds: string[];
+}
+
+export interface VenueEventsData {
+  isEventsLoading: boolean;
+  isError: boolean;
+
+  events: WithVenueId<WithId<VenueEvent>>[];
+}
+
+export const useVenueEvents: ReactHook<VenueEventsProps, VenueEventsData> = ({
+  venueIds,
+}) => {
+  const { events, isError, isLoading } = useEventsContext();
+
+  const requestedEvents = useMemo(
+    () => events.filter((event) => venueIds.includes(event.venueId)),
+    [events, venueIds]
+  );
+
+  return {
+    events: requestedEvents,
+    isError: isError,
+    isEventsLoading: isLoading,
+  };
+};
+
+export interface EventsContextState {
+  events: WithVenueId<WithId<VenueEvent>>[];
+  isLoading: boolean;
+  isError: boolean;
+  venueEventsError?: Error;
+  sovereignVenueIdError?: string;
+}
+
+const EventsContext = createContext<EventsContextState | undefined>(undefined);
+
+export interface EventsProviderProps {
+  venueId?: string;
+}
+
+export const EventsProvider: React.FC<EventsProviderProps> = ({
+  children,
+  venueId,
+}) => {
+  const {
+    sovereignVenueId,
+    isSovereignVenueIdLoading,
+
+    errorMsg: sovereignVenueIdError,
+  } = useSovereignVenueId({
+    venueId,
+  });
```

Comment:
`EventsProvider` is setup to need `venueId` passed to it. Yet where you instantiate it, you're not passing `venueId` to it at all?

Reply:
Seems this was fixed in https://github.com/sparkletown/sparkle/pull/1631